File: backend/src/main/java/com/example/backend/global/auth/service/CookieService.java
package com.example.backend.global.auth.service;

import org.springframework.stereotype.Service;

import com.example.backend.global.auth.util.CookieUtil;
import com.example.backend.global.auth.util.JwtUtil;

import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import lombok.RequiredArgsConstructor;

/**
 * CookieService
 * 쿠키 관련 로직을 처리하는 클래스
 * @author 100mi
 */
@Service
@RequiredArgsConstructor
public class CookieService {

	private final CookieUtil cookieUtil;
	private final JwtUtil jwtUtil;

	public void addAccessTokenToCookie(String accessToken, HttpServletResponse response) {
		cookieUtil.addTokenToCookie("accessToken", accessToken,
			jwtUtil.getAccessTokenExpirationTime(), response);
	}

	public void addRefreshTokenToCookie(String refreshToken, HttpServletResponse response) {
		cookieUtil.addTokenToCookie("refreshToken", refreshToken,
			jwtUtil.getRefreshTokenExpirationTime(), response);
	}

	public String getAccessTokenFromCookie(HttpServletRequest request) {
		return cookieUtil.getTokenFromCookie("accessToken", request);
	}

	public String getRefreshTokenFromCookie(HttpServletRequest request) {
		return cookieUtil.getTokenFromCookie("refreshToken", request);
	}

	public void clearTokenFromCookie(HttpServletResponse response) {
		cookieUtil.addTokenToCookie("accessToken", null, 0, response);
		cookieUtil.addTokenToCookie("refreshToken", null, 0, response);
	}
}

File: backend/src/main/java/com/example/backend/global/auth/service/CookieService.kt
package com.example.backend.global.auth.service

import com.example.backend.global.auth.util.CookieUtil
import com.example.backend.global.auth.util.JwtUtil
import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse
import org.springframework.stereotype.Service
import java.util.concurrent.TimeUnit

/**
 * CookieService
 * 쿠키 관련 로직을 처리하는 클래스
 * @author 100mi
 */
@Service
class CookieService(
    private val cookieUtil: CookieUtil,
    private val jwtUtil: JwtUtil
) {

    fun addAccessTokenToCookie(accessToken: String, response: HttpServletResponse) {
        cookieUtil.addTokenToCookie(
            "accessToken", accessToken,
            TimeUnit.MILLISECONDS.toSeconds(jwtUtil.getAccessTokenExpirationTime()),
            response
        )
    }

    fun addRefreshTokenToCookie(refreshToken: String, response: HttpServletResponse) {
        cookieUtil.addTokenToCookie(
            "refreshToken", refreshToken,
            TimeUnit.MILLISECONDS.toSeconds(jwtUtil.getRefreshTokenExpirationTime()),
            response
        )
    }

    fun addRefreshTokenToCookieWithSameSiteNone(refreshToken: String, response: HttpServletResponse) {
        cookieUtil.addTokenToCookieWithSameSiteNone(
            "refreshToken", refreshToken,
            TimeUnit.MILLISECONDS.toSeconds(jwtUtil.getRefreshTokenExpirationTime()), response
        )
    }

    fun getAccessTokenFromCookie(request: HttpServletRequest): String? {
        return cookieUtil.getTokenFromCookie("accessToken", request)
    }

    fun getRefreshTokenFromCookie(request: HttpServletRequest): String? {
        return cookieUtil.getTokenFromCookie("refreshToken", request)
    }

    fun clearTokenFromCookie(response: HttpServletResponse) {
        cookieUtil.addTokenToCookie("accessToken", "", 0, response)
        cookieUtil.addTokenToCookie("refreshToken", "", 0, response)
    }
}
